rust/src/arrow.rs

```diff
@@ -439,6 +424,68 @@ impl RecordBatchExt for RecordBatch {
     }
 }
 
+fn merge(left_fields: Vec<FieldRef>,
+         left_columns: Vec<ArrayRef>,
+         right_fields: Vec<FieldRef>,
+         right_columns: Vec<ArrayRef>) -> (Vec<FieldRef>, Vec<ArrayRef>) {
```

Comment:
is `Vec<FieldRef> + Vec<ArrayRef>` to be a RecordBatch?

Reply:
yes, RecordBatch constructor takes list of tuple (field, array ref)